src/org/opendatakit/briefcase/model/BriefcasePreferences.java

```diff
@@ -21,39 +21,133 @@
+ * {@link java.util.prefs.Preferences Preferences API}.
+ */
 public class BriefcasePreferences {
-
+  
   public static final String VERSION = "v1.4.9 Production";
+  public static final String USERNAME = "username";
+  public static final String TOKEN = "token";
+  public static final String AGGREGATE_1_0_URL = "url_1_0";
+  public static final String AGGREGATE_0_9_X_URL = "url_0_9_X";
+  
+  private static final String BRIEFCASE_DIR_PROPERTY = "briefcaseDir";
+  
+  static {
+    // load the security provider
+    Security.addProvider(new BouncyCastleProvider());
+  }
+
+  private final Preferences preferences;
+  
+  private BriefcasePreferences(Class<?> node, PreferenceScope scope) {
+    this.preferences = scope.preferenceFactory(node);
+  }
+  
+  /**
+   * Factory that returns the application scoped <tt>BriefcasePreferences</tt> object.
+   * The method always return the same instance.
+   */
+  public static BriefcasePreferences applicationScoped() {
+    return Preference.APPLICATION_SCOPED;
+  }
+  
+  /**
+   * Factory that creates a class scoped <tt>BriefcasePreferences</tt> object. 
+   * @param node the managing class
+   */
+  public static BriefcasePreferences forClass(Class<?> node) {
+    return new BriefcasePreferences(node, PreferenceScope.CLASS_NAME);
+  }
+  
+  /**
+   * Returns the value associated with the specified key in this preference
+   * node. Returns the specified default if there is no value associated with
+   * the key, or the backing store is inaccessible.
+   * 
+   * @param key
+   *          key whose associated value is to be returned.
+   * @param defaultValue
+   *          the value to be returned in the event that this preference node
+   *          has no value associated with key.
+   * @return the value associated with key, or defaultValue if no value is associated
+   *         with key, or the backing store is inaccessible.
+   */
+  public String get(String key, String defaultValue) {
+    return preferences.get(key, defaultValue);
+  }
+  
+  /**
+   * Associates the specified value with the specified key in this preference
+   * node.
+   * 
+   * @param key
+   *          key with which the specified value is to be associated.
+   * @param value
+   *          value to be associated with the specified key.
+   */
+  public void put(String key, String value) {
+    preferences.put(key, value);
+  }
+  
+  /**
+   * Removes the value associated with the specified key in this preference
+   * node, if any.
+   * 
+   * @param key
+   *          key whose mapping is to be removed from the preference node.
+   */
+  public void remove(String key) {
+    preferences.remove(key);
+  }
 
   public static void setBriefcaseDirectoryProperty(String value) {
-    if ( value == null ) {
-      getApplicationPreferences().remove(BRIEFCASE_DIR_PROPERTY);
+    if (value == null) {
+      Preference.APPLICATION_SCOPED.remove(BRIEFCASE_DIR_PROPERTY);

```

Comment:
You're right, I wrote it to provide access to the <tt>Preference.APPLICATION_SCOPED</tt> instance from other classes. But since we're only accessing the instance through its specific methods, we don't need the factory.